SQLQuery_LibraryBooksBD.sql

```diff
@@ -0,0 +1,842 @@
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[UsersAndBooks_ExistByUserId]
+@UserId int
+AS
+BEGIN
+	SELECT TOP 1 1 FROM UsersAndBooks 
+	WHERE UserId = @UserId
+END
+
+GO
+/****** Object:  StoredProcedure [dbo].[UsersAndBooks_GetBooksByUserId]    Script Date: 19.07.2021 19:13:04 ******/
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[UsersAndBooks_GetBooksByUserId]
+@UserId int
+AS
+BEGIN
+SELECT Book.Id as Id, 
+		Book.Title as Title,
+		Book.AuthorId as AuthorId,
+		Author.FullName as AuthorFullName,
+		Book.Page as Page,
+		Book.Description as Description
+FROM UsersAndBooks
+INNER JOIN Book ON UsersAndBooks.BookId = Book.Id
+INNER JOIN Author ON Book.AuthorId = Author.Id
+WHERE UsersAndBooks.UserId = @UserId
+END
+
+GO
+/****** Object:  StoredProcedure [dbo].[UsersAndBooks_GetBooksUnAvailableByUserId]    Script Date: 19.07.2021 19:13:04 ******/
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[UsersAndBooks_GetBooksUnAvailableByUserId]
+@UserId int
+AS
+BEGIN
+SELECT Book.Id as Id, 
+		Book.Title as Title,
+		Book.AuthorId as AuthorId,
+		Author.FullName as AuthorFullName,
+		Book.Page as Page,
+		Book.Description as Description
+FROM  Book
+INNER JOIN Author ON Book.AuthorId = Author.Id
+LEFT JOIN 
+(
+	SELECT Book.Id as Id
+	FROM UsersAndBooks
+	INNER JOIN Book ON UsersAndBooks.BookId = Book.Id
+	WHERE UsersAndBooks.UserId = @UserId
+)	AS BooksByUserId
+ON BooksByUserId.Id = Book.Id
+WHERE BooksByUserId.Id is NULL
+END
+
+GO
+/****** Object:  StoredProcedure [dbo].[UsersAndBooks_GetUsersByBookId]    Script Date: 19.07.2021 19:13:04 ******/
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[UsersAndBooks_GetUsersByBookId]
+@BookId int
+AS
+BEGIN
+SELECT UserInfo.UserId as UserId, 
+		UserInfo.SurName as SurName,
+		UserInfo.FirstName as FirstName,
+		UserInfo.SecondName as SecondName,
+		UserInfo.DateOfBirth as DateOfBirth,
+		UserInfo.Phone as Phone
+FROM UsersAndBooks
+INNER JOIN UserInfo ON UsersAndBooks.UserId = UserInfo.UserId
+WHERE UsersAndBooks.BookId = @BookId
+END
+
+GO
+/****** Object:  StoredProcedure [dbo].[UsersAndBooks_GetUsersUnAvailableByBookId]    Script Date: 19.07.2021 19:13:04 ******/
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[UsersAndBooks_GetUsersUnAvailableByBookId]
+@BookId int
+AS
+BEGIN
+SELECT UserInfo.UserId as UserId, 
+		UserInfo.SurName as SurName,
+		UserInfo.FirstName as FirstName,
+		UserInfo.SecondName as SecondName,
+		UserInfo.DateOfBirth as DateOfBirth,
+		UserInfo.Phone as Phone
+FROM UserInfo
+LEFT JOIN 
+(
+SELECT UserInfo.UserId as UserId
+FROM UsersAndBooks
+INNER JOIN UserInfo ON UsersAndBooks.UserId = UserInfo.UserId
+WHERE UsersAndBooks.BookId = @BookId
+) AS UsersByBookId ON UsersByBookId.UserId = UserInfo.UserId
+WHERE UsersByBookId.UserId is NULL
+END
```

Comment:
Indentation helps readablility in these complex queries
```suggestion
BEGIN
    SELECT 
        UserInfo.UserId as UserId, 
        UserInfo.SurName as SurName,
        UserInfo.FirstName as FirstName,
        UserInfo.SecondName as SecondName,
        UserInfo.DateOfBirth as DateOfBirth,
        UserInfo.Phone as Phone
    FROM UserInfo
        LEFT JOIN (
            SELECT UserInfo.UserId as UserId
            FROM UsersAndBooks
                INNER JOIN UserInfo ON UsersAndBooks.UserId = UserInfo.UserId
            WHERE UsersAndBooks.BookId = @BookId
        ) AS UsersByBookId ON UsersByBookId.UserId = UserInfo.UserId
    WHERE UsersByBookId.UserId is NULL
END
```